dipy/denoise/noise_estimate.py

```diff
@@ -203,15 +188,21 @@ def piesno_3D(data, N, alpha=0.01, l=100, itermax=100, eps=1e-5):
     else:
         q = 0.5
 
-    # prevent overflow in sum_m2
-    data = data.astype(np.float32)
-
     # Initial estimation of sigma
     denom = np.sqrt(2 * _inv_nchi_cdf(N, 1, q))
     m = np.percentile(data, q * 100) / denom
+
+    # if the percentile is 0, then more than half of the slice is zero and give up
+    if m == 0:
```

Comment:
This seems to be the one most substantial change here. Under what conditions does this happen? I don't have much experience using this code, so I need as gentle an explanation you can muster here.